packages/odyssey-react-mui/src/labs/DataTable.tsx

```diff
@@ -480,28 +501,139 @@ const DataTable = ({
+
+          const { key } = event;
+          const isSpaceKey = key === " ";
+          const isEnterKey = key === "Enter";
+          const isEscapeKey = key === "Escape";
+
+          if (isEscapeKey) {
+            resetDraggingAndHoveredRow(table);
+            return;
+          }
+
+          const isSpaceOrEnter = isSpaceKey || isEnterKey;
+
+          if (isSpaceOrEnter) {
+            event.preventDefault();
+          }
+
+          const isArrowDown = key === "ArrowDown";
+          const isArrowUp = key === "ArrowUp";
+
+          const currentIndex = row.index + (page - 1) * resultsPerPage;
+
+          if (draggingRow) {
+            if (hoveredRow && hoveredRow.index) {
+              const { index } = hoveredRow;
+
+              if (isSpaceOrEnter && index != currentIndex) {
+                handleReordering({
+                  rowId: row.id,
+                  newIndex: index,
+                });
+
+                setTimeout(() => {
+                  resetDraggingAndHoveredRow(table);
+                }, 1000);
+                return;
+              }
+
+              if (isArrowDown) {
+                const nextRowFromDataAfterDownArrow = data[index + 1];
+
+                if (nextRowFromDataAfterDownArrow) {
+                  getRowFromTableAndSetHovered(
+                    table,
+                    nextRowFromDataAfterDownArrow.id
+                  );
+                }
+              }
 
-        setDraggingRow(null);
-      },
+              if (isArrowUp) {
+                const nextRowFromDataAfterUpArrow = data[index - 1];
 
-      onDragCapture: () => {
-        if (!draggingRow && table.getState().draggingRow?.id) {
-          setDraggingRow(table.getState().draggingRow);
-        }
-      },
+                if (nextRowFromDataAfterUpArrow) {
+                  getRowFromTableAndSetHovered(
+                    table,
+                    nextRowFromDataAfterUpArrow.id
+                  );
+                }
+              }
+            } else {
+              if (isArrowDown || isArrowUp) {
+                if (isArrowDown) {
+                  const shouldDisableReOrder =
+                    currentIndex === page * resultsPerPage - 1;
+                  if (shouldDisableReOrder) {
+                    return;
+                  }
+                }
+
+                if (isArrowUp) {
+                  const shouldDisableReOrder =
+                    currentIndex === 0 ||
+                    currentIndex === (page - 1) * resultsPerPage;
+                  if (shouldDisableReOrder) {
+                    return;
+                  }
+                }
+
+                const nextIndex = isArrowDown ? row.index + 1 : row.index - 1;
+                const nextRowFromData = data[nextIndex];
+
+                if (nextRowFromData) {
+                  getRowFromTableAndSetHovered(table, nextRowFromData.id);
+                }
+              }
+            }
```

Comment:
Style nit: Consider extracting this out to an arrow press function to make the parent decision block of code easier to read.